registry/storage/layerwriter.go

```diff
@@ -14,9 +14,11 @@ import (
 	"github.com/docker/docker/pkg/tarsum"
 )
 
-// layerUploadController is used to control the various aspects of resumable
+var _ distribution.LayerUpload = &layerWriter{}
```

Comment:
I would agree if this were just a writer. While `fileReader` and `fileWriter` just implement reader/writer functionality, this higher-level abstraction includes a digest and other functionality.

We can leave this as you have it here, since this will be refactored into `blobWriter` in the near future.